src/netCDF4/__init__.pyi

```diff
@@ -0,0 +1,542 @@
+
+import os
+from typing import (Any, Callable, Final, Generic, Iterable, Literal, Mapping,
+                    NoReturn, Self, Sequence, TypeAlias, TypeVar, Union, overload)
+
+import numpy as np
+import numpy.typing as npt
+from cftime import date2index, date2num, num2date
```

Comment:
Rather than import these (since cftime doesn't provide stubs) we can stub them out as follows:

```python
# date2index, date2num, and num2date are actually provided by cftime and if stubs for
# cftime are completed these should be removed.
def date2index(
    dates: (datetime.datetime | cftime.datetime | Sequence[datetime.datetime | cftime.datetime] | DateTimeArr),
    nctime: Variable,
    calendar: CfCalendar | None = None,
    select: Literal["exact", "before", "after", "nearest"] = "exact",
    has_year_zero: bool | None = None,
) -> int | npt.NDArray[np.int_]: ...
def date2num(
    dates: (datetime.datetime | cftime.datetime | Sequence[datetime.datetime | cftime.datetime] | DateTimeArr),
    units: str,
    calendar: CfCalendar | None = None,
    has_year_zero: bool | None = None,
    longdouble: bool = False,
) -> np.number | npt.NDArray[np.number]: ...
def num2date(
    times: Sequence[int | float | np.number] | npt.NDArray[np.number],
    units: str,
    calendar: CfCalendar = "standard",
    only_use_cftime_datetimes: bool = True,
    only_use_python_datetimes: bool = False,
    has_year_zero: bool | None = None,
) -> datetime.datetime | DateTimeArr: ...
```